controllers/storageautoscaler/storageautoscaler_controller.go

```diff
@@ -67,6 +67,31 @@ func (r *StorageAutoscalerReconciler) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	// if more than one storage autoscaler is present in the cluster with same device class and same storage cluster
+	// return error
+	storageAutoScalerList := &ocsv1.StorageAutoScalerList{}
+	err = r.List(ctx, storageAutoScalerList,
+		client.InNamespace(request.Namespace),
+		client.MatchingFields{"spec.storageCluster.name": storageAutoScaler.Spec.StorageCluster.Name, "spec.deviceClass": storageAutoScaler.Spec.DeviceClass},
+	)
+	if err != nil {
+		r.Log.Error(err, "failed to list storage autoscaler")
+		return reconcile.Result{}, err
+	}
+
+	if len(storageAutoScalerList.Items) > 1 {
+		// update the status
+		originalStorageAutoScaler := storageAutoScaler.DeepCopy()
+		err := r.updateStatus(ctx, storageAutoScaler, originalStorageAutoScaler)
```

Comment:
Don't we need to update the CR status after applying the error state?

Reply:
Yes, but we need to introduce a new `Invalid`  phase will do it as a follow up pr